test/over_react/util/prop_conversion_test.dart

```diff
@@ -0,0 +1,1067 @@
+}
+
+//
+// Test components for use as JS `component` prop
+//
+
+mixin ExpectsDartMapPropProps on UiProps {
+  Map dartMapProp;
+}
+
+UiFactory<ExpectsDartMapPropProps> ExpectsDartMapProp = uiForwardRef(
+  (props, ref) {
+    verifyType<Map>(props.dartMapProp);
+
+    return (Dom.div()
+      ..addProps(props)
+      // Use the prop in a way that would fail if it wasn't the correct type
+      ..['data-dart-map-prop'] = jsonEncode(props.dartMapProp)
+      ..ref = ref)(props.children);
+  },
+  _$ExpectsDartMapPropConfig, // ignore: undefined_identifier
+);
+
+mixin ExpectsDartStylePropProps on UiProps {}
+
+UiFactory<ExpectsDartStylePropProps> ExpectsDartStyleProp = uiForwardRef(
+  (props, ref) {
+    verifyType<Map<String, dynamic>>(props.style);
+
+    return (Dom.div()
+      ..addProps(props)
+      ..style = {
+        // Use the prop in a way that would fail if it wasn't the correct type
+        ...props.style,
+      })(props.children);
+  },
+  _$ExpectsDartStylePropConfig, // ignore: undefined_identifier
+);
+
+mixin ExpectsListChildrenPropProps on UiProps {}
+
+UiFactory<ExpectsListChildrenPropProps> ExpectsListChildrenProp = uiForwardRef(
+  (props, ref) {
+    verifyType<List>(props.children);
+
+    return (Dom.div()..addProps(props))(
+      // Use the prop in a way that would fail if it wasn't the correct type
+      props.children.map((child) => child),
+    );
+  },
+  _$ExpectsListChildrenPropConfig, // ignore: undefined_identifier
+);
+
+//
+// Other test components
+//
+
+UiFactory<ClassComponentProps> ClassComponent =
+    castUiFactory(_$ClassComponent); // ignore: undefined_identifier
+
+mixin ClassComponentProps on UiProps {}
+
+class ClassComponentComponent extends UiComponent2<ClassComponentProps> {
+  @override
+  render() {
+    return (Dom.div()..addProps(props))(props.children);
+  }
+}
+
+mixin BasicForwardRefProps on UiProps {}
+
+const basicForwardRefAttribute = 'data-basic-forward-ref';
+UiFactory<BasicForwardRefProps> BasicForwardRef = uiForwardRef(
+  (props, ref) {
+    return (Dom.span()
+      ..addProps(props)
+      ..[basicForwardRefAttribute] = ''
+      ..ref = ref)(props.children);
+  },
+  _$BasicForwardRefConfig, // ignore: undefined_identifier
+);
+
+mixin DartTestJsWrapperPropsMixin on UiProps {
+  void Function(DartTestJsWrapperProps props) onRender;
+}
+
+class DartTestJsWrapperProps = UiProps
+    with TestJsProps, DartTestJsWrapperPropsMixin;
+
+UiFactory<DartTestJsWrapperProps> DartTestJsWrapper = uiForwardRef(
+  (props, ref) {
+    props.onRender?.call(props);
+
+    final consumedProps =
+        props.staticMeta.forMixins({DartTestJsWrapperPropsMixin});
+    return (TestJs()
+      ..addUnconsumedProps(props, consumedProps)
+      ..ref = ref)(props.children);
+  },
+  _$DartTestJsWrapperConfig, // ignore: undefined_identifier
+);
+
+//
+// OverReact bindings for a JS test component, used as a MUI component analogue
+// to assert correct behavior in more functional-style cases.
```

Comment:
#nit this reference to MUI is no longer relevant
```suggestion
// OverReact bindings for a JS test component
// to assert correct behavior in more functional-style cases.
```